File: scala2-materials/exercises/Language/src/main/scala/videostore/CommandLine.java
/*
 * This example adapted from 'Refactoring' by Martin Fowler
 *  
 */
package videostore;

public class CommandLine {
    public static void main(String[] args) {
        Movie peterPan = new Movie("Peter Pan", PriceCode.CHILDRENS);
        Movie theHulk = new Movie("The Hulk", PriceCode.REGULAR);
        Movie starWars = new Movie("Star Wars", PriceCode.REGULAR);
        Movie toyStory = new Movie("Toy Story", PriceCode.CHILDRENS);
        Movie killBill = new Movie("Kill Bill", PriceCode.NEW_RELEASE);
        Rental r1 = new Rental(peterPan, 2);
        Rental r2 = new Rental(theHulk, 1);
        Rental r3 = new Rental(starWars, 3);
        Rental r4 = new Rental(toyStory, 2);
        Rental r5 = new Rental(killBill, 4);
        Customer customer = new Customer("Joe Bloggs");
        customer.addRental(r1);
        customer.addRental(r2);
        customer.addRental(r3);
        customer.addRental(r4);
        customer.addRental(r5);
        System.out.println(customer.statement());
    }
}

File: scala2-materials/solutions/Language/src/main/scala/videostore/CommandLine.scala
package videostore

import PriceCode._

object CommandLine {
  def main(args: Array[String]) {
    val peterPan = new Movie("Peter Pan", CHILDRENS)
    val theHulk = new Movie("The Hulk", REGULAR)
    val starWars = new Movie("Star Wars", REGULAR)
    val toyStory = new Movie("Toy Story", CHILDRENS)
    val killBill = new Movie("Kill Bill", NEW_RELEASE)

    val r1 = new Rental(peterPan, 2)
    val r2 = new Rental(theHulk, 1)
    val r3 = new Rental(starWars, 3)
    val r4 = new Rental(toyStory, 2)
    val r5 = new Rental(killBill, 4)

    val customer = new Customer("Joe Bloggs")
    customer.addRental(r1)
    customer.addRental(r2)
    customer.addRental(r3)
    customer.addRental(r4)
    customer.addRental(r5)

    println(customer.statement)
  }
}
